flink-kubernetes-operator-autoscaler/src/main/java/org/apache/flink/kubernetes/operator/autoscaler/ScalingMetricCollector.java

```diff
@@ -184,7 +184,13 @@ protected JobTopology queryJobTopology(RestClusterClient<String> restClient, Job
                                             JobDetailsInfo.JobVertexDetailsInfo
                                                     ::getMaxParallelism));
 
-            return JobTopology.fromJsonPlan(jobDetailsInfo.getJsonPlan(), maxParallelismMap);
+            // Flink 1.17 introduced a strange behaviour where the JobDetailsInfo#getJsonPlan()
+            // doesn't return the jsonPlan correctly but it wraps with RawJson{json='plan'}
+            var rawPlan = jobDetailsInfo.getJsonPlan();
+            var json =
```

Comment:
This strongly depends on the exact string we get from the `rawPlan`. No whitespaces, no anything. How sure we about this? Could it change with different json libraries, or whatever?

Reply:
This logic is fixed based on the Flink client library (1.17.1). This issue will be fixed in 1.17.2 then we can remove this workaround.